util/signing/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+/// A collection of bytes.
+#[derive(Debug, Clone)]
+pub struct Bytes(pub Vec<u8>);
+
+/// A signature.
+#[derive(Debug, Clone)]
+pub struct Signature(pub Bytes);
+
+/// A public key.
+#[derive(Debug, Clone)]
+pub struct PublicKey(pub Bytes);
+
+/// Version of a key.
+/// Default mean the current key.
+#[derive(Debug, Clone, Default)]
+pub struct KeyVersion(pub String);
+
+/// Id that identify a Key.
+#[derive(Debug, Clone)]
+pub struct KeyId(pub String);
+
+pub struct SigningService;
+
+impl SigningService {
```

Comment:
Should this rather be a trait? I think we need implementations of this for the different backends.

Reply:
The idea was to be backend agnostic so that the application don't have to know how keys are organized.